### ISSUE TITLE
Allow passing a custom optimizer for adversarial training

### ISSUE THREAD

**Issue by iamgroot42** [tensorflow/cleverhans#433]
The subroutines currently used for adversarial training (as given in the demo [here](https://github.com/tensorflow/cleverhans/blob/master/cleverhans_tutorials/mnist_tutorial_tf.py)) currently use an optimizer [hard-coded](https://github.com/tensorflow/cleverhans/blob/master/cleverhans/utils_tf.py#L117) into the code. The specified optimizer (or learning rate) may not necessarily work well for some models. For instance, I tried running adversarial training for SVHN on a Resnet-110 and did not cross 11% or so validation accuracy. 

I can submit a pull request for the same, if someone else isn't working on it already.

**Comment by iamgroot42** [comment]
The code doesn't support batch normalization or regularizers currently either, which is quite problematic.

**Comment by npapernot** [comment]
thanks for raising this issue! It looks like some of these limitations will be addressed by #423 at least in part. Feel free to comment on the current code
